shell/platform/darwin/ios/platform_message_handler_ios.mm

```diff
@@ -127,8 +124,7 @@ - (void)dispatch:(dispatch_block_t)block {
   message_handlers_.erase(channel);
   if (handler) {
     message_handlers_[channel] = {
-        .task_queue = fml::scoped_nsprotocol(
-            [static_cast<NSObject<FlutterTaskQueueDispatch>*>(task_queue) retain]),
+        .task_queue = (NSObject<FlutterTaskQueueDispatch>*)task_queue,
```

Comment:
This is possibly unsafe.  You are storing an objc pointer inside of a c++ data structure and relying on ARC to keep it alive.  I'm like 60% sure this isn't okay, maybe it's fine if the structs and c++ data structures are only ever used from objc++ with ARC?

If you haven't already, you could write a tiny sample program to demonstrate that it works.  You should probably add a comment if it is safe, you could even throw in a the link to the gist that proves it's safe.  You may still have to use a smart pointer that uses bridge directives to manually do the reference counting.

Reply:
> maybe it's fine if the structs and c++ data structures are only ever used from objc++ with ARC?

It is, but since the struct is part of the public part of the header and doesn't assert ARC I agree that this isn't currently safe.

Let's leave this as a `scoped_nsprotocol` for now, and unwind it when the ARC migration is complete and it's easier to reason about the safety.

Reply:
Thanks for keeping this on the rails, @gaaclarke.  I am out of my depth with the parts of this migration that touch C++.